src/merkle/merkle_tree.rs

```diff
@@ -0,0 +1,278 @@
+            match sibling_hash {
+                SiblingsHash::LeftSibling(left_hash) => {
+                    left_hash.hash(&mut hasher);
+                    transaction.hash(&mut hasher);
+                }
+                SiblingsHash::RightSibling(right_hash) => {
+                    transaction.hash(&mut hasher);
+                    right_hash.hash(&mut hasher);
+                }
+            }
+
+            transaction = hasher.finish();
+        }
+
+        transaction == self.merkle_root.hash_value
+    }
+
+    fn recursive_get_proof(
+        actual_node: &MerkleNode,
+        proof: &mut Vec<SiblingsHash>,
+        transaction_hash: u64,
+    ) -> bool {
+        if let Some(left) = &actual_node.left_son {
+            if left.hash_value == transaction_hash {
+                if let Some(right_sibling) = &actual_node.right_son {
+                    proof.push(SiblingsHash::RightSibling(right_sibling.hash_value));
+                }
+                return true;
+            }
+            if Self::recursive_get_proof(left, proof, transaction_hash) {
+                if let Some(right_sibling) = &actual_node.right_son {
+                    proof.push(SiblingsHash::RightSibling(right_sibling.hash_value));
+                }
+                return true;
+            }
+        }
+
+        if let Some(right) = &actual_node.right_son {
+            if right.hash_value == transaction_hash {
+                if let Some(left_sibling) = &actual_node.left_son {
+                    proof.push(SiblingsHash::LeftSibling(left_sibling.hash_value));
+                }
+                return true;
+            }
+
+            if Self::recursive_get_proof(right, proof, transaction_hash) {
+                if let Some(left_sibling) = &actual_node.left_son {
+                    proof.push(SiblingsHash::LeftSibling(left_sibling.hash_value));
+                }
+                return true;
+            }
+        }
+        false
+    }
+
+    pub fn get_proof(&mut self, transaction: H) -> Vec<SiblingsHash> {
+        let mut proof = Vec::new();
+        let mut hasher = DefaultHasher::new();
+        transaction.hash(&mut hasher);
+        Self::recursive_get_proof(&self.merkle_root, &mut proof, hasher.finish());
+
+        proof
+    }
+
+    pub fn add(&mut self, transaction: H) {
+        self.leafs.push(transaction);
+        let mut leafs = Vec::new();
+        for leaf in &self.leafs {
+            leafs.push(leaf.clone());
+        }
+        self.merkle_root = Self::create_tree(leafs.clone()).merkle_root;
+    }
+}
+
+#[cfg(test)]
+pub mod test {
+
+    use crate::merkle::merkle_tree::MerkleTree;
+
+    #[test]
+    fn a_new_merkle_tree_contains_nothing() {
+        let transactions: Vec<String> = Vec::new();
+        let merkle_tree = MerkleTree::new(transactions);
+
+        assert!(merkle_tree.leafs.is_empty());
+    }
+
+    #[test]
+    fn a_merkle_tree_can_contains_one_transaction() {
```

Comment:
```suggestion
    fn a_merkle_tree_can_contain_one_transaction() {
```
Just a typo, the same with the following test names.